Database

Greg-2600 opened the pull request.

Tr777ple commented:
Is this still for oobir.com stock system or no? Just curious, forgot what I
subscribed to, lol

If so, can I get a login? Wanting to purchase some stock that is suggested
by AI/the code you wrote/automated etc...definitely have an opportunity to
make some money. If not, no worries man.

Hope you and the family are doing good brother :)

On Tue, Oct 17, 2023, 10:21 PM dependabot[bot] ***@***.***>
wrote:

> Bumps urllib3 <https://github.com/urllib3/urllib3> from 1.26.9 to 1.26.18.
> Release notes
>
> *Sourced from urllib3's releases
> <https://github.com/urllib3/urllib3/releases>.*
>
> 1.26.18
>
>    - Made body stripped from HTTP requests changing the request method to
>    GET after HTTP 303 "See Other" redirect responses. (GHSA-g4mx-q9vg-27p4
>    <https://github.com/advisories/GHSA-g4mx-q9vg-27p4>)
>
> 1.26.17
>
>    - Added the Cookie header to the list of headers to strip from
>    requests when redirecting to a different host. As before, different headers
>    can be set via Retry.remove_headers_on_redirect. (GHSA-v845-jxx5-vc9f
>    <https://github.com/advisories/GHSA-v845-jxx5-vc9f>)
>
> 1.26.16
>
>    - Fixed thread-safety issue where accessing a PoolManager with many
>    distinct origins would cause connection pools to be closed while requests
>    are in progress (#2954
>    <https://redirect.github.com/urllib3/urllib3/issues/2954>)
>
> 1.26.15
>
>    - Fix socket timeout value when HTTPConnection is reused (
>    urllib3/urllib3#2645
>    <https://redirect.github.com/urllib3/urllib3/issues/2645>)
>    - Remove "!" character from the unreserved characters in IPv6 Zone ID
>    parsing (urllib3/urllib3#2899
>    <https://redirect.github.com/urllib3/urllib3/issues/2899>)
>    - Fix IDNA handling of 'x80' byte (urllib3/urllib3#2901
>    <https://redirect.github.com/urllib3/urllib3/issues/2901>)
>
> 1.26.14
>
>    - Fixed parsing of port 0 (zero) returning None, instead of 0 (#2850
>    <https://redirect.github.com/urllib3/urllib3/issues/2850>)
>    - Removed deprecated HTTPResponse.getheaders() calls in urllib3.contrib
>    module.
>
> 1.26.13
>
>    - Deprecated the HTTPResponse.getheaders() and HTTPResponse.getheader()
>    methods.
>    - Fixed an issue where parsing a URL with leading zeroes in the port
>    would be rejected even when the port number after removing the zeroes was
>    valid.
>    - Fixed a deprecation warning when using cryptography v39.0.0.
>    - Removed the <4 in the Requires-Python packaging metadata field.
>
> 1.26.12
>
>    - Deprecated the urllib3[secure] extra and the
>    urllib3.contrib.pyopenssl module. Both will be removed in v2.x. See
>    this GitHub issue
>    <https://redirect.github.com/urllib3/urllib3/issues/2680> for
>    justification and info on how to migrate.
>
> 1.26.11
>
> *If you or your organization rely on urllib3 consider supporting us via
> GitHub Sponsors <https://github.com/sponsors/urllib3>.*
>
> ⚠️ *urllib3 v2.0 will drop support for Python 2*: Read more in the v2.0
> Roadmap <https://urllib3.readthedocs.io/en/latest/v2-roadmap.html>
>
>    - Fixed an issue where reading more than 2 GiB in a call to
>    HTTPResponse.read would raise an OverflowError on Python 3.9 and earlier.
>
> 1.26.10
>
> *If you or your organization rely on urllib3 consider supporting us via
> GitHub Sponsors <https://github.com/sponsors/urllib3>.*
>
> ⚠️ *urllib3 v2.0 will drop support for Python 2*: Read more in the v2.0
> Roadmap <https://urllib3.readthedocs.io/en/latest/v2-roadmap.html>
>
> 🔐 *This is the first release to be signed with Sigstore!* You can verify
> the distributables using the .sig and .crt files included on this release.
>
>    - Removed support for Python 3.5
>    - Fixed an issue where a ProxyError recommending configuring the proxy
>    as HTTP instead of HTTPS could appear even when an HTTPS proxy wasn't
>    configured.
>
> Changelog
>
> *Sourced from urllib3's changelog
> <https://github.com/urllib3/urllib3/blob/main/CHANGES.rst>.*
>
> 1.26.18 (2023-10-17)
>
>    - Made body stripped from HTTP requests changing the request method to
>    GET after HTTP 303 "See Other" redirect responses.
>
> 1.26.17 (2023-10-02)
>
>    - Added the Cookie header to the list of headers to strip from
>    requests when redirecting to a different host. As before, different headers
>    can be set via Retry.remove_headers_on_redirect. ([#3139](
>    https://github.com/urllib3/urllib3/issues/3139) <
>    https://github.com/urllib3/urllib3/pull/3139>_)
>
> 1.26.16 (2023-05-23)
>
>    - Fixed thread-safety issue where accessing a PoolManager with many
>    distinct origins would cause connection pools to be closed while requests
>    are in progress ([#2954](https://github.com/urllib3/urllib3/issues/2954)
>    <https://github.com/urllib3/urllib3/pull/2954>_)
>
> 1.26.15 (2023-03-10)
>
>    - Fix socket timeout value when HTTPConnection is reused ([#2645](
>    https://github.com/urllib3/urllib3/issues/2645) <
>    https://github.com/urllib3/urllib3/issues/2645>__)
>    - Remove "!" character from the unreserved characters in IPv6 Zone ID
>    parsing ([#2899](https://github.com/urllib3/urllib3/issues/2899) <
>    https://github.com/urllib3/urllib3/issues/2899>__)
>    - Fix IDNA handling of '\x80' byte ([#2901](
>    https://github.com/urllib3/urllib3/issues/2901) <
>    https://github.com/urllib3/urllib3/issues/2901>__)
>
> 1.26.14 (2023-01-11)
>
>    - Fixed parsing of port 0 (zero) returning None, instead of 0. (
>    [#2850](https://github.com/urllib3/urllib3/issues/2850) <
>    https://github.com/urllib3/urllib3/issues/2850>__)
>    - Removed deprecated getheaders() calls in contrib module. Fixed the
>    type hint of PoolKey.key_retries by adding bool to the union. ([#2865](
>    https://github.com/urllib3/urllib3/issues/2865) <
>    https://github.com/urllib3/urllib3/issues/2865>__)
>
> 1.26.13 (2022-11-23)
>
>    - Deprecated the HTTPResponse.getheaders() and HTTPResponse.getheader()
>    methods.
>    - Fixed an issue where parsing a URL with leading zeroes in the port
>    would be rejected even when the port number after removing the zeroes was
>    valid.
>    - Fixed a deprecation warning when using cryptography v39.0.0.
>    - Removed the <4 in the Requires-Python packaging metadata field.
>
> 1.26.12 (2022-08-22)
>
>    - Deprecated the urllib3[secure] extra and the
>    urllib3.contrib.pyopenssl module. Both will be removed in v2.x. See
>    this GitHub issue <https://github.com/urllib3/urllib3/issues/2680>_
>    for justification and info on how to migrate.
>
> 1.26.11 (2022-07-25)
>
>    - Fixed an issue where reading more than 2 GiB in a call to
>    HTTPResponse.read would
>
> ... (truncated)
> Commits
>
>    - 9c2c230
>    <https://github.com/urllib3/urllib3/commit/9c2c2307dd1d6af504e09aac0326d86ee3597a0b>
>    Release 1.26.18 (#3159
>    <https://redirect.github.com/urllib3/urllib3/issues/3159>)
>    - b594c5c
>    <https://github.com/urllib3/urllib3/commit/b594c5ceaca38e1ac215f916538fb128e3526a36>
>    Merge pull request from GHSA-g4mx-q9vg-27p4
>    <https://github.com/advisories/GHSA-g4mx-q9vg-27p4>
>    - 944f0eb
>    <https://github.com/urllib3/urllib3/commit/944f0eb134485f41bc531be52de12ba5a37bca73>
>    [1.26] Use vendored six in urllib3.contrib.securetransport
>    - c9016bf
>    <https://github.com/urllib3/urllib3/commit/c9016bf464751a02b7e46f8b86504f47d4238784>
>    Release 1.26.17
>    - 0122035
>    <https://github.com/urllib3/urllib3/commit/01220354d389cd05474713f8c982d05c9b17aafb>
>    Backport GHSA-v845-jxx5-vc9f
>    <https://github.com/advisories/GHSA-v845-jxx5-vc9f> (#3139
>    <https://redirect.github.com/urllib3/urllib3/issues/3139>)
>    - e63989f
>    <https://github.com/urllib3/urllib3/commit/e63989f97d206e839ab9170c8a76e3e097cc60e8>
>    Fix installing brotli extra on Python 2.7
>    - 2e7a24d
>    <https://github.com/urllib3/urllib3/commit/2e7a24d08713a0131f0b3c7197889466d645cc49>
>    [1.26] Configure OS for RTD to fix building docs
>    - 57181d6
>    <https://github.com/urllib3/urllib3/commit/57181d6ea910ac7cb2ff83345d9e5e0eb816a0d0>
>    [1.26] Improve error message when calling urllib3.request() (#3058
>    <https://redirect.github.com/urllib3/urllib3/issues/3058>)
>    - 3c01480
>    <https://github.com/urllib3/urllib3/commit/3c0148048a523325819377b23fc67f8d46afc3aa>
>    [1.26] Run coverage even with failed jobs
>    - d94029b
>    <https://github.com/urllib3/urllib3/commit/d94029b7e2193ff47b627906a70e06377a09aae8>
>    Release 1.26.16
>    - Additional commits viewable in compare view
>    <https://github.com/urllib3/urllib3/compare/1.26.9...1.26.18>
>
>
> [image: Dependabot compatibility score]
> <https://docs.github.com/en/github/managing-security-vulnerabilities/about-dependabot-security-updates#about-compatibility-scores>
>
> Dependabot will resolve any conflicts with this PR as long as you don't
> alter it yourself. You can also trigger a rebase manually by commenting @dependabot
> rebase.
> ------------------------------
> Dependabot commands and options
>
> You can trigger Dependabot actions by commenting on this PR:
>
>    - @dependabot rebase will rebase this PR
>    - @dependabot recreate will recreate this PR, overwriting any edits
>    that have been made to it
>    - @dependabot merge will merge this PR after your CI passes on it
>    - @dependabot squash and merge will squash and merge this PR after
>    your CI passes on it
>    - @dependabot cancel merge will cancel a previously requested merge
>    and block automerging
>    - @dependabot reopen will reopen this PR if it is closed
>    - @dependabot close will close this PR and stop Dependabot recreating
>    it. You can achieve the same result by closing it manually
>    - @dependabot show <dependency name> ignore conditions will show all
>    of the ignore conditions of the specified dependency
>    - @dependabot ignore this major version will close this PR and stop
>    Dependabot creating any more for this major version (unless you reopen the
>    PR or upgrade to it yourself)
>    - @dependabot ignore this minor version will close this PR and stop
>    Dependabot creating any more for this minor version (unless you reopen the
>    PR or upgrade to it yourself)
>    - @dependabot ignore this dependency will close this PR and stop
>    Dependabot creating any more for this dependency (unless you reopen the PR
>    or upgrade to it yourself)
>    You can disable automated security fix PRs for this repo from the Security
>    Alerts page <https://github.com/Greg-2600/oobir/network/alerts>.
>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/Greg-2600/oobir/pull/17
> Commit Summary
>
>    - 57392fb
>    <https://github.com/Greg-2600/oobir/pull/17/commits/57392fba918e91d677def3ba27d2bf8825f47d81>
>    Bump urllib3 from 1.26.9 to 1.26.18
>
> File Changes
>
> (1 file <https://github.com/Greg-2600/oobir/pull/17/files>)
>
>    - *M* requirements.txt
>    <https://github.com/Greg-2600/oobir/pull/17/files#diff-4d7c51b1efe9043e44439a949dfd92e5827321b34082903477fd04876edb7552>
>    (2)
>
> Patch Links:
>
>    - https://github.com/Greg-2600/oobir/pull/17.patch
>    - https://github.com/Greg-2600/oobir/pull/17.diff
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Greg-2600/oobir/pull/17>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AZZNT3QDEO4NYLAPWIG2UCDX75DLFAVCNFSM6AAAAAA6EZWTP2VHI2DSMVQWIX3LMV43ASLTON2WKOZRHE2DQNZRGM2DGNA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

Greg-2600 committed:
Docs: document PostgreSQL caching and cache management endpoints

Greg-2600 committed:
Scripts: preserve Ollama model volume on undeploy; Docs: clarify deploy/undeploy behavior

Greg-2600 committed:
API: orient balance sheet and income statement JSON by index (dates) for UI

Greg-2600 committed:
Add comprehensive UI enhancements and database caching

- Enhanced stock results page header with horizontal price trend summary
- Added company summary box with business description and key details
- Improved technical indicators chart with volume and oscillators (RSI, MACD, Stochastic, ADX)
- Fixed option chain rendering to display formatted table
- Made OOBIR logo clickable to return to landing page
- Added database caching to all data and AI endpoints for improved performance
- Hide empty data cards to reduce clutter
- Display sector/industry information in stock header

Greg-2600 committed:
Update tests with database cache mocking and enhance README documentation

- Add @patch decorators for db.get_cached_data and db.set_cached_data to all endpoint tests
- Verify cache writes with assert_called_once() in all successful test cases
- Update README with database caching features and UI enhancements
- Document horizontal price trend, company summary box, and clickable logo
- Fold test update documentation into README testing section
- Remove redundant TEST_UPDATES.md file
- Bump version to 1.2.0

Greg-2600 committed:
Fix model inconsistency and clean up code

- Update OLLAMA_MODEL in docker-compose.yml to huihui_ai/llama3.2-abliterate:3b (matches flow.py)
- Remove duplicate comment in flow_api.py with_ai_cache function
- Remove debug console.log from web/app.js for production
- Make web/config.js dynamic: auto-detects host instead of hardcoded IP
- Delete temporary pr_body.md file

Greg-2600 committed:
Refactor README: Clean up formatting, add database to architecture diagrams

- Remove malformed code from Technology Principles section
- Update Container Architecture diagram to include PostgreSQL
- Enhance System Architecture diagram with comprehensive detail
- Reorganize Caching Infrastructure section for better clarity
- Professional formatting with aligned ASCII diagrams
- All diagrams now show complete technology stack

Greg-2600 committed:
Remove duplicate model pull from deploy script

- Removed llama3.2:1b pull (using huihui_ai/llama3.2-abliterate:3b only)
- Consistent with docker-compose.yml OLLAMA_MODEL config
